doc/users-guide/datapar/promotion.rst

```diff
@@ -0,0 +1,324 @@
+:title: Users Guide: Promotion
+
+.. _ug-promotion:
+
+Promotion: implicit data parallelism
+====================================
+
+In Chapel, *function promotion* (or simply *promotion*) refers to an
+implicit form of data parallelism that is triggered by passing a
+collection of values to an argument that expects a single value.  More
+precisely, when an array of type *t* (or a range or domain whose index
+type is *t*) is passed to a function argument of type *t*, that
```

Comment:
I think it would be great to just say "ranges, domains, arrays, or forall expressions over these types". I think it would be good to include forall expressions in this document in some capacity too (I can't remember right now if you already did).

I also think we should make an issue about how more general promotion should probably be supported (I don't think I was aware of the missing functionality).